import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="4.11.0" severity="performance" hasFix />

`Border.all` constructor calls **const** `Border.fromBorderSide` constructor under the hood. This rule allows to replace `Border.all` with **const** `Border.fromBorderSide`.

### Example {#example}

**❌ Bad:**

```dart
class BorderWidget extends StatelessWidget {
  final Widget child;

  const RoundedWidget({
    Key? key,
    required this.child,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      //LINT
      border: Border.all(
        color: const Color(0xFF000000),
        width: 1.0,
        style: BorderStyle.solid,
      ),
      child: child,
    );
  }
}
```

**✅ Good:**

```dart
class BorderWidget extends StatelessWidget {
  final Widget child;

  const RoundedWidget({
    Key? key,
    required this.child,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      border: const Border.fromBorderSide(BorderSide(
        color: const Color(0xFF000000),
        width: 1.0,
        style: BorderStyle.solid,
      )),
      child: child,
    );
  }
}
```
