import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="4.17.0" severity="warning" hasFix />

The rule detects `Expanded` widgets that contain empty `SizedBox/Container` and proposes to replace them with the `Spacer` widget.

### Example {#example}

**❌ Bad:**

```dart
Column(
    children: [
        Container(),
        const Expanded(child: SizedBox()),
        Container(),
    ]
)
```

**✅ Good:**

```dart
Column(
    children: [
        Container(),
        const Spacer(),
        Container(),
    ]
)
```
