import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="5.1.0" severity="style" hasConfig hasFix  />

Enforces named argument order in function and constructor invocations to be the same as corresponding named parameter declaration order.

The rule can be used together with [sort_child_properties_last](https://dart-lang.github.io/linter/lints/sort_child_properties_last.html) rule by setting `child-last: true` in rule configuration.

### ⚙️ Config example {#config-example}

```yaml
dart_code_metrics:
  ...
  rules:
    ...
    - arguments-ordering:
        child-last: true
```

### Example {#example}

**❌ Bad:**

```dart
Person buildPerson({ String name, String surname, String age });

// LINT
final person = buildPerson(age: 42, surname: 'The Pooh', name: 'Winnie');
```

**✅ Good:**

```dart
Person buildPerson({ String name, String surname, String age });

// LINT
final person = buildPerson(name: 'Winnie', surname: 'The Pooh', age: 42);
```
