import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="4.7.0" severity="warning" />

Warns when a enum constant is missing in a map declaration.

### Example {#example}

**❌ Bad:**

```dart
enum SomeEnum {
  firstEntry,
  secondEntry,
  thirdEntry,
}

extension SomeX on SomeEnum {
  // LINT
  static const firstMap = <SomeEnum, String>{
    CountyCode.firstEntry: 'foo',
    CountyCode.secondEntry: 'bar',
  };

  // LINT twice since `secondEntry` and `thirdEntry` are missing
  static const secondMap = <SomeEnum, String>{
    CountyCode.firstEntry: 'foo',
  };
}
```

**✅ Good:**

```dart
enum SomeEnum {
  firstEntry,
  secondEntry,
  thirdEntry,
}

extension SomeX on SomeEnum {
  static const firstMap = <SomeEnum, String>{
    CountyCode.firstEntry: 'foo',
    CountyCode.secondEntry: 'bar',
    CountyCode.thirdEntry: 'baz',
  };

  static const secondMap = <SomeEnum, String>{
    CountyCode.firstEntry: 'foo',
    CountyCode.secondEntry: 'bar',
    CountyCode.thirdEntry: 'baz',
  };
}
```
