import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="5.2.0" severity="warning" hasFix />

Checks for unnessesary conditional expressions.

### Example {#example}

**❌ Bad:**

```dart
bool baz() => foo ? true : false; // LINT

bool bar() => foo ? false : true; // LINT
```

**✅ Good:**

```dart
bool baz() => foo;

bool bar() => !foo;
```
