import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="4.17.0" severity="warning" />

Warns when a file has multiple `exports` declarations with the same URI.

### Example {#example}

**❌ Bad:**

```dart
import "package:flutter/material.dart";
import "package:flutter_bloc/flutter_bloc.dart";

export 'package:my_app/good_folder/something.dart';
export 'package:my_app/good_folder/something.dart'; // LINT
```

**✅ Good:**

```dart
import "package:flutter/material.dart";
import "package:flutter_bloc/flutter_bloc.dart";

export 'package:my_app/good_folder/something.dart';
```
