import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="1.7.0" severity="style" hasFix />

Warns when a literal value is on the left hand side in a binary expressions.

**❌ Bad:**

```dart
final a = 1 + b;
```

**✅ Good:**

```dart
final a = b + 1;
```

Inspired by [TSLint rule](https://palantir.github.io/tslint/rules/binary-expression-operand-order/)
