import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="1.5.1" severity="style" />

Enforces blank line between statements and return in a block.

### Example {#example}

**❌ Bad:**

```dart
  if ( ... ) {
    ...
    return ...; // LINT
  }
```

**✅ Good:**

```dart
  if ( ... ) {
    return ...;
  }

  if ( ... ) {
    ...

    return ...;
  }
```
