import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="1.5.0" severity="style" hasFix />

Warns on comparison to a boolean literal, as in `x == true`. Comparing boolean values to boolean literals is unnecessary, as those expressions will result in booleans too. Just use the boolean values directly or negate them.

### Example {#example}

**❌ Bad:**

```dart
  var b = x == true; // LINT
  var c = x != true; // LINT

   // LINT
  if (x == true) {
    ...
  }

   // LINT
  if (x != false) {
    ...
  }
```

**✅ Good:**

```dart
  var b = x;
  var c = !x;

  if (x) {
    ...
  }

  if (!x) {
    ...
  }
```
