import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="4.19.0" severity="warning" hasFix />

Warns when `List.from()` factory is used instead of `List.of()`.

The difference between `List.of()` and `List.from()` is that `.of()` takes an argument of the same type as what it returns and enforces it at compilation time, and that `.from()` allows potentially unsafe downcasting and enforces convertibility at runtime.

Additional resources:

- <https://github.com/dart-lang/linter/issues/2555>
- <https://github.com/dart-lang/linter/issues/2066>

### Example {#example}

**❌ Bad:**

```dart
...
var intList = [1, 2, 3];

var copy = List<int>.from(intList); // LINT
var numList = List<num>.from(intList); // LINT

var unspecifedList = List.from(intList); // LINT
```

**✅ Good:**

```dart
var numList = <num>[1, 2, 3];

var intList = List<int>.from(numList);
```
