import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="1.8.0" severity="style" hasFix hasConfig />

Recommends to use a conditional expression instead of assigning to the same thing or return statement in each branch of an if statement.

Use `ignore-nested` configuration, if you want to ignore cases with multiple nested conditional expressions (default is false).

### ⚙️ Config example {#config-example}

```yaml
dart_code_metrics:
  ...
  rules:
    ...
    - prefer-conditional-expressions:
        ignore-nested: true
```

### Example {#example}

**❌ Bad:**

```dart
  int a = 0;

  // LINT
  if (a > 0) {
    a = 1;
  } else {
    a = 2;
  }

  // LINT
  if (a > 0) a = 1;
  else a = 2;

  int function() {
    // LINT
    if (a == 1) {
        return 0;
    } else {
        return 1;
    }

    // LINT
    if (a == 2) return 0;
    else return 1;
  }
```

**✅ Good:**

```dart
  int a = 0;

  a = a > 0 ? 1 : 2;

  int function() {
    return a == 2 ? 0 : 1;
  }
```
