import RuleDetails from '@site/src/components/RuleDetails';

<RuleDetails version="4.19.0" severity="warning" />

Warns if the file within `/test` contains a `main`, but the file name doesn't end with `_test.dart`.

:::note

If you want to set `exclude` config for this rule, the default `['lib/**', 'bin/**']` will be overriden.

:::

### Example {#example}

**❌ Bad:**

File name: **some_file.dart**

```dart
void main() {
  ...
}
```

**✅ Good:**

File name: **some_file_test.dart**

```dart
void main() {
  ...
}
```

File name: **some_other_file.dart**

```dart
void helperFunction() {
  ...
}
```
